src/client/static/hydration.tsx

```diff
@@ -15,8 +17,21 @@ export const hydrateApp = () => {
 
   const clientState = routingConfig.clientState;
 
-  const { abTesting: { mvtId = 0, forcedTestVariants = {} } = {} } =
-    clientState;
+  const {
+    abTesting: { mvtId = 0, forcedTestVariants = {} } = {},
+    sentryConfig: { stage, build, dsn },
+  } = clientState;
+
+  // Disable Sentry client side logging during local development.
+  if (stage !== 'DEV') {
```

Comment:
rather than disabling on dev, I would suggest enable it if the `dsn` is supplied, as we'd want to replicate as much as possible between DEV/CODE/PROD